### PR TITLE
Add more milestone info to the release doc

### PR DESCRIPTION
**Pull request by jeefy**
First pass at addressing https://github.com/kubernetes/sig-release/issues/320. 

Feedback and nits appreciated. :)

### CODE REVIEW

File: contributors/devel/sig-release/release.md [hunk 1]
```diff
@@ -136,6 +136,22 @@ feature work for the given release has been defined in suitable
 planning artifacts in conjunction with the Release Team's [enhancements
 lead](https://git.k8s.io/sig-release/release-team/role-handbooks/enhancements/README.md).
 
+After enhancement freeze, applying milestones to PRs and Issues is
+important. Items within the milestone are used as a punchdown list to
+complete the release. Milestones **must** be applied correctly and managed
+aggressively to properly gauge what will go into a release and what
+effort may be needed and when.
+
+What would qualify to go into a currenet milestone? Any item that is
```
Comment (geekygirldawn) [inline contributors/devel/sig-release/release.md]: Little typo in current :)

Comment (jeefy) [reply contributors/devel/sig-release/release.md]: Derp thanks!

File: contributors/devel/sig-release/release.md [hunk 2]
```diff
@@ -136,6 +136,22 @@ feature work for the given release has been defined in suitable
...
+What would qualify to go into a current milestone? Any item that is
+required for, or slated to go into, the current release. These could
+be issues or PRs in repos including (but not limited to):
+
+- kubernetes/enhancements
+- kubernetes/kubernetes
+- kubernetes/release
+- kubernetes/sig-release
+- kubernetes/website
+
```
Comment (guineveresaenger) [inline contributors/devel/sig-release/release.md]: I am curious - do all patches get milestoned these days? I'd love some clarifying language here!

**Comment by guineveresaenger** [comment]
this is good to have!
/approve

**Comment by tpepper** [comment]
This looks good to me, but I'll hold off with the lgtm command to get additional review.

File: contributors/devel/sig-release/release.md (hunk 2 above)
Comment (tpepper) [reply contributors/devel/sig-release/release.md]: Also cherry-pick patches must have the milestone associated with the branch to which they are targeted. I've proposed some additional detail on this front via #4081 .

**Comment by guineveresaenger** [comment]
Let's reword, encompassing the following points (just trying to capture my best understanding of the situation)

- **on issues** milestones must be applied correctly, via triage by the SIG, so that rel team can follow bugs and enhancements (any enhancement related issue needs a milestone)
- **on PRs against master** milestones are auto-applied so this is less necessary
- **on PRS against anything not master** the milestone requirement is absolute and iron-clad necessary

TL;DR: milestone early, milestone often, and everyone will be happier. 😛

**Comment by guineveresaenger** [comment]
I wasn't going for using my somewhat casual language here 😅 but I think it's clear, and I like it.

/lgtm

**Comment by guineveresaenger** [comment]
@justaugustus or @tpepper for approval?

File: contributors/devel/sig-release/release.md (hunk 2 above)
Comment (tpepper) [reply contributors/devel/sig-release/release.md]: Given the automation on release branches, users never are interacting with the milestone.  If you make a PR against release-1.16 branch, it is milestoned as 1.16.  I think it clutters the documentation to have separate treatment then.  If it's worded as:

```
**On PRs against the master branch**, milestones are auto-applied so
this is less necessary. 

**On PRS against anything not master** milestones are auto-applied so
this is less necessary.
```

Which then can be condensed to:

```
** On PRs**, milestones are auto-applied so
this is less necessary.
```

Which I don't think is the desire here.

I think what we actually have is:

 * At creation time, PRs against the master branch need humans to hint at which milestone they might want the PR to target.
 * Once merged, PRs against the master branch have milestones auto-applied so from that time onward
human management of that PR's milestone is less necessary. 
 * On PRs against anything not master branch, milestones are auto-applied when the PR is created so
no human management of the milestone is less ever necessary.

**Comment by guineveresaenger** [comment]
@jeefy - PTAL?

**Comment by jeefy** [comment]
@guineveresaenger Thanks for the ping. Updated w/ @tpepper prose. :)

**Comment by justaugustus** [comment]
Thanks y'all!
/lgtm
/approve
/milestone v1.17
